adapter/pkg/messaging/azure_listener.go

```diff
@@ -0,0 +1,160 @@
+/*
+ *  Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+// Package messaging holds the implementation for event listeners functions
+package messaging
+
+import (
+	logger "github.com/wso2/product-microgateway/adapter/internal/loggers"
+	servicebus "github.com/Azure/azure-service-bus-go"
+	"context"
+	"strings"
+)
+
+
+func startBrokerConsumer(topicName string, ns *servicebus.Namespace,
+	availableTopicList []*servicebus.TopicEntity, componentName string) {
+
+	var topicExistForFurtherProcess bool
+	var subscriptionExistForFurtherProcess bool
+	parentContext := context.Background()
+	logger.LoggerMgw.Info("[TEST][FEATURE_FLAG_REPLACE_EVENT_HUB] Starting broker consumer for topic name : " +
+		topicName)
+
+	//In Azure service bus subscription names can contain letters, numbers, periods (.), hyphens (-), and
+	// underscores (_), up to 50 characters. Subscription names are also case-insensitive.
+	var subscriptionName = topicName + "_" + componentName + "_sub"
```

Comment:
When multiple adapters starting up, using the same subscription name would cause issues right?

Reply:
When multiple adapters are there then for each messages will be delivered in round robin manner

Reply:
As per Viraj do we need a broadcast behaviour here ?

Reply:
We can have a uuid or ask the deployment to give a id as env variable and create a new subscription so that each adapther will get same msg